### PR TITLE
[doc] added infix notation for constructors

### LINKED ISSUE #4426: [doc] Include infix notation for constructors
I think it would be good to include in [Mixfix Operators](https://agda.readthedocs.io/en/latest/language/mixfix-operators.html) that the infix notation can also be used for constructors.

If you have a good example to illustrate it or comments, please post them below.

### PR DESCRIPTION
**Pull request by pnlph**
See issue #4426

### CODE REVIEW

**Review by UlfNorell — APPROVED** [review]

File: doc/user-manual/language/mixfix-operators.lagda.rst [hunk 1]
```diff
@@ -17,23 +17,43 @@
 Mixfix Operators
 ****************
 
-A name containing one or more name parts and one or more ``_`` can be used as an operator where the arguments go in place of the ``_``. For instance, an application of the name ``if_then_else_`` to arguments ``x``, ``y``, and ``z`` can be written either as a normal application ``if_then_else_ x y z`` or as an operator application ``if x then y else z``.
-
-Examples:
+A type name, function name, or constructor name can comprise one or more name
+parts if we separate them with one or more underscore characters ``_``, and the
```
Comment (andreasabel) [inline doc/user-manual/language/mixfix-operators.lagda.rst]: > if we separate them with one or more underscore characters

No, only with exactly one underscore!  Better formulation:
> if we separate them with underscore characters

File: doc/user-manual/language/mixfix-operators.lagda.rst [hunk 2]
```diff
@@ -17,23 +17,43 @@
...
+
+For instance, we can join with underscores the name parts ``if``, ``then``,
+and ``else`` into a single name ``if_then_else_``. The application of the
+function name ``if_then_else_`` to some arguments named ``x``, ``y``, and ``z``
+can still be written either as:
+
+* a standard application by using the full name ``if_then_else_ x y z``
+* an operator application by placing the arguments between the name parts
+  ``if x then y else z``, leaving a space between arguments and part names
+
```
Comment (andreasabel) [inline doc/user-manual/language/mixfix-operators.lagda.rst:41]: While you are at it, you could include _sections_. All alternatives are:
```agda
if x then y else z
(if_then y else z) x
(if x then_else z) y
if x then y else_ z
if x then_else_ y z
if_then y else_ x z
(if_then_else z) x y
if_then_else_ x y z
```
Please add some of the section alternatives (leaving some underscores).

File: doc/user-manual/language/mixfix-operators.lagda.rst [hunk 3]
```diff
@@ -17,23 +17,43 @@
...
   if false then x else y = y
 
-  _⇒_   : Bool → Bool → Bool
-  true  ⇒ b = b
-  false ⇒ _ = true
-
+  -- Example constructor name _::_
+  data List (A : Set) : Set where
+    nil  : List A
+    _::_ : A -> List A -> List A
```
Comment (andreasabel) [inline doc/user-manual/language/mixfix-operators.lagda.rst]: Please use the unicode arrow here (and preferable the unicode ::).

**Review by andreasabel — CHANGES_REQUESTED** [review]
Thanks for the contribution.
Got a few suggestions for improvement.

**Review by andreasabel — APPROVED** [review]
Thanks for fixing!

**Pull request merged**